### PR TITLE
add forward compatibility and deprecation notices

### PR DESCRIPTION
**Pull request by Ener-Getick**
This PR adds BC layers for #1107.

### CODE REVIEW

File: Inflector/DoctrineInflector.php [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+use Doctrine\Common\Inflector\Inflector;
+
+/**
+ * Inflector object using the Doctrine/Inflector.
+ *
+ * @author Mark Kazemier <Markkaz>
+ */
+class DoctrineInflector implements InflectorInterface
+{
+    public function __construct()
```
Comment (sstok) [inline Inflector/DoctrineInflector.php]: Is this empty constructor really needed?

Comment (Ener-Getick) [inline Inflector/DoctrineInflector.php]: Not yet I forgot to extend the old class which triggers a deprecation in its constructor.

File: Tests/Negotiation/FormatNegotiatorTest.php [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+namespace FOS\RestBundle\Tests\Negotiation;
+
+class FormatNegotiatorTest extends \PHPUnit_Framework_TestCase
+{
+    const FORMAT_NEGOTIATOR_CLASS = 'FOS\RestBundle\Negotiation\FormatNegotiator';
+
+    protected $negotiator;
+
+    public function setUp()
```
Comment (sstok) [inline Tests/Negotiation/FormatNegotiatorTest.php]: FYI setUp() is actually defined protected in the parent class ;)

**Comment by Ener-Getick** [comment]
@xabbuh @lsmith77 we should decide if we create bc layers for 1.* or if we leave it as is.
As @stof said [here](https://github.com/FriendsOfSymfony/FOSRestBundle/pull/1171#issuecomment-159972323) IMO it's easier for everyone to upgrade a project to 2.0 with BC layers but that's also true that we lose time by creating them and that we could use this time to create new features...

IMO we should at least create BC layers for the features the most used (context, annotation properties, ...).

**Comment by lsmith77** [comment]
should go to the 1.8 branch /cc @stof

**Comment by lsmith77** [comment]
@stof can you modify the target branch? would like to see this merged quickly .. @Ener-Getick alternatively you can also just open a new PR

**Comment by Ener-Getick** [comment]
I opened #1254

**Pull request closed without merging**